.github/actions/spell-check/candidate.patterns

```diff
@@ -1,23 +1,36 @@
 # marker to ignore all code on line
 ^.*/\* #no-spell-check-line \*/.*$
-# marker for ignoring a comment to the end of the line
-// #no-spell-check.*$
+# marker to ignore all code on line
+^.*\bno-spell-check(?:-line|)(?:\s.*|)$
+
+# https://cspell.org/configuration/document-settings/
+# cspell inline
+^.*\b[Cc][Ss][Pp][Ee][Ll]{2}:\s*[Dd][Ii][Ss][Aa][Bb][Ll][Ee]-[Ll][Ii][Nn][Ee]\b
 
 # patch hunk comments
 ^\@\@ -\d+(?:,\d+|) \+\d+(?:,\d+|) \@\@ .*
 # git index header
-index [0-9a-z]{7,40}\.\.[0-9a-z]{7,40}
+index (?:[0-9a-z]{7,40},|)[0-9a-z]{7,40}\.\.[0-9a-z]{7,40}
+
+# file permissions
+['"`\s][-bcdLlpsw](?:[-r][-w][-Ssx]){2}[-r][-w][-SsTtx]\+?['"`\s]
+
+# css url wrappings
+\burl\([^)]+\)
 
 # cid urls
 (['"])cid:.*?\g{-1}
 
 # data url in parens
-\(data:[^)]*?(?:[A-Z]{3,}|[A-Z][a-z]{2,}|[a-z]{3,})[^)]*\)
+\(data:(?:[^) ][^)]*?|)(?:[A-Z]{3,}|[A-Z][a-z]{2,}|[a-z]{3,})[^)]*\)
 # data url in quotes
-([`'"])data:.*?(?:[A-Z]{3,}|[A-Z][a-z]{2,}|[a-z]{3,}).*\g{-1}
+([`'"])data:(?:[^ `'"].*?|)(?:[A-Z]{3,}|[A-Z][a-z]{2,}|[a-z]{3,}).*\g{-1}
 # data url
 data:[-a-zA-Z=;:/0-9+]*,\S*
 
+# https/http/file urls
+#(?:\b(?:https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]
+
 # mailto urls
 mailto:[-a-zA-Z=;:/?%&0-9+@.]{3,}
```

Comment:
Oh, hah. Well, I guess I hadn't run into it as a problem. I've added it to spell-check-this so in some future refresh it'll arrive.

Reply:
Realistically, that'd have been fine here because `jay` and `something` are likely to be valid words -- the heuristics here are mostly to catch non-words. -- The actual regex to match an email address is thousands of characters long and isn't worth applying to a case like this. -- But, I do sympathize as I get annoyed when systems reject `+` in the user portion of email addresses (something I use regularly, both personally and for work).

Reply:
Oh yeah, I've had systems accept + in e-mail addresses but then some subsystems inside that company wouldn't and that would create all sort of issues for my account.